services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/functionaltests/document/APISurfaceAreaReview.java

```diff
@@ -0,0 +1,401 @@
+
+        assertThat(customDoc.toJson()).isEqualTo("{\"direct_attr\": \"sample_value\",\"custom_attr\": {\"aBoolean\": false,"
+                                                 + "\"string\": \"str\",\"longNumber\": 25},\"customO_list\": [{\"aBoolean\": "
+                                                 + "false,\"string\": \"str_one\",\"longNumber\": 26}, {\"aBoolean\": true,"
+                                                 + "\"string\": \"str_two\",\"longNumber\": 27}]}");
+
+
+        // Extracting Custom Object\
+        CustomClassForDocumentAPI customAttr = customDoc.get("custom_attr", EnhancedType.of(CustomClassForDocumentAPI.class));
+        System.out.println("customAttr " +customAttr);
+        assertThat(customAttr).isEqualTo(customObject);
+
+
+        // Extracting custom List
+        List<CustomClassForDocumentAPI> extractedList = customDoc.getList("customO_list",
+                                                                        EnhancedType.of(CustomClassForDocumentAPI.class));
+
+        assertThat(extractedList).isEqualTo(Arrays.asList(customObjectOne,customObjectTwo));
+
+    }
+
+
+    /**
+     * ERROR Case when Attribute Converters are not supplied and a custom Object is added
+     */
+    @Test
+    void attributeConverter_Not_Added(){
+
+        CustomClassForDocumentAPI customObject = getCustomObject("str", 25L, false);
+
+        assertThatExceptionOfType(IllegalStateException.class)
+            .isThrownBy(() ->
+                EnhancedDocument.builder()
+                                .addString("direct_attr", "sample_value")
+                                .add("custom_attr", customObject)
+                                .build()
+
+            ).withMessage("Converter not found for EnhancedType(software.amazon.awssdk.enhanced.dynamodb.converters"
+                                      + ".document.CustomClassForDocumentAPI)");
+
+    }
+
+
+    @Test
+    void putSimpleDocumentWithSimpleValues() {
+        table.putItem(EnhancedDocument
+                          .builder()
+                          .addString(HASH_KEY, "first_hash")
+                          .addString(SORT_KEY, "1_sort")
+                          .addNull("null_Key")
+                          .addBoolean("boolean_key", true)
+                          .addNumber("number_int", 2)
+                          .addNumber("number_SdkNumber", SdkNumber.fromString("5"))
+                          .addSdkBytes("SdkBytes", SdkBytes.fromUtf8String("sample_binary"))
+                          .build());
+
+        assertThat(
+            table.getItem(EnhancedDocument.builder()
+                                          .add(SORT_KEY, "1_sort")
+                                          .addString(HASH_KEY, "first_hash").build()).get(SORT_KEY)).isEqualTo("1_sort");
+
+
+    }
+
+    /**
+     *- Does EnhancedDocument.Builder needs add(key, Object, attributeConverterProvider)
```

Comment:
Yeah I like this idea. How would it affect the table schema or table objects? Will they be updated?